acdc/acdc_utils.py

```diff
@@ -181,9 +183,18 @@ def kl_divergence(
         assert kl_div.shape == mask_repeat_candidates.shape, (kl_div.shape, mask_repeat_candidates.shape)
         denom = mask_repeat_candidates.long().sum()
         kl_div = kl_div * mask_repeat_candidates
-        answer = kl_div.sum() / denom
+        if return_one_element:
+            answer = kl_div.sum() / denom
+        else:
+            answer = kl_div
     else:
-        answer = kl_div.mean()
+        if return_one_element:
+            answer = kl_div.mean()
+        else:
+            answer = kl_div
+
+    if not return_tensor:
```

Comment:
Seems easier to just call `.item()` on the output of `kl_divergence` if it's needed